sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/Common/AbstractExplodedGraph.cs

```diff
@@ -274,6 +275,15 @@ protected virtual void VisitSimpleBlock(SimpleBlock block, ExplodedGraphNode nod
 
         protected ProgramState CleanStateAfterBlock(ProgramState programState, Block block)
         {
+            // We do not clean state when entering a using block statement, as some variables declared with "using" keywords
+            // may still be implictly disposed later, even if they are not referenced anymore. State will still be cleaned
+            // in the "VisitSimpleBlock" of UsingEndBlock
+            if (block is JumpBlock jumpBlock &&
+                jumpBlock.SuccessorBlock is UsingEndBlock)
+            {
```

Comment:
Should not differentiate here between the using statement and local declaration with using?

Reply:
Local declaration with using will not create a `UsingEndBlock` at all, the way the CFG is built was not modified. Here, only the rule was updated to look for local using declarations to for "dispose" calls or its equivalent